### PR TITLE
[FIX] test_http: adapt test to succeed with ubuntu 24.04 noble

### PR DESCRIPTION
**Pull request by d-fence**
The brand new Ubuntu Noble 24.04 ships python3-werkzeug 3.0.1. This versions has a bug [0] which leads to this test failing as double slashes get fixed by werkzeug even though `merge_slashes` parameter is set to False by default in Odoo.

This should not have a big impact on real life but the test should be kept to ensure that the actual behavior of a 404 is not broken outside of Ubuntu Noble.

So, with this commit, if the impacted version of werkzeug is used the test will not fail with a fixed url and a redirect response.

The issue that exists from 2.2.0 up to 3.0.1 is fixed upstream [1] in 3.0.2.

A bug report was filled in Ubuntu [2] (Debian is not impacted)

[0]: https://github.com/pallets/werkzeug/issues/2834
[1]: https://github.com/pallets/werkzeug/pull/2860
[2]: https://bugs.launchpad.net/ubuntu/+source/python-werkzeug/+bug/2066041

### CODE REVIEW

File: odoo/addons/test_http/tests/test_misc.py [hunk 1]
```diff
@@ -20,7 +21,13 @@
 class TestHttpMisc(TestHttpBase):
     def test_misc0_redirect(self):
         res = self.nodb_url_open('/test_http//greeting')
-        self.assertEqual(res.status_code, 404)
+        awaited_codes = [404]
+        werkzeug_version = metadata.version('werkzeug')
+        if parse_version(werkzeug_version) >= parse_version('2.0.2') and parse_version(werkzeug_version) <= parse_version('3.0.1'):
+            # Bug in werkzeug from 2.2.0 up to 3.0.1 (shipped in Ubuntu Noble 24.04)
+            # Not a really big deal but should be remvoved once fixed upstream
```
Comment (Julien00859) [inline odoo/addons/test_http/tests/test_misc.py:28]:
```suggestion
        werkzeug_version = metadata.version('werkzeug')
        if parse_version('2.0.2') <= parse_version(werkzeug_version) <= parse_version('3.0.1'):
            # Bug in werkzeug from 2.2.0 up to 3.0.1 (shipped in Ubuntu Noble 24.04)
            # not a big deal but should be removed once fixed upstream.
```

**Review by Julien00859 — APPROVED** [review]
@robodoo delegate+

File: odoo/addons/test_http/tests/test_misc.py (hunk 1 above)
Comment (Julien00859) [reply odoo/addons/test_http/tests/test_misc.py:28]: also change parse_version('2.0.2')

**Review by Julien00859 — CHANGES_REQUESTED** [review]
[`importlib.metadata`](https://docs.python.org/3/library/importlib.metadata.html) is 3.8 but odoo16 must support 3.7

**Comment by d-fence** [comment]
> [`importlib.metadata`](https://docs.python.org/3/library/importlib.metadata.html) is 3.8 but odoo16 must support 3.7

Nice catch (unlike parse_version('2.0.2') which was not really your catch)

**Review by Julien00859 — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**